### PR TITLE
Visuals for tower

### PR DESCRIPTION
**Pull request by Perrytheplatypus43**

### CODE REVIEW

File: visuals.js [hunk 1]
```diff
@@ -56,6 +56,9 @@ module.exports = class Visuals {
             if (VISUALS.TERMINAL) {
                 Visuals.terminal(room);
             }
+            if (VISUALS.TOWER) {
+ +				room.structures.towers.forEach(t => Visuals.drawTowerInfo(t));
+ +			}
```
Comment (Spedwards) [inline visuals.js]: Weird pluses and tabs instead of spaces.

File: visuals.js [hunk 2]
```diff
@@ -277,20 +280,26 @@ module.exports = class Visuals {
             vis.circle(weakest.pos.x, weakest.pos.y, {radius: 0.4, fill: '#FF0000', opacity: 0.3, strokeWidth: 0,});
             let y = weakest.pos.y - 0.5; // base y pos - consistent with spawns, labs, and controllers
             const look = weakest.pos.lookFor(LOOK_STRUCTURES);
-            const spawns = _.find(look, o => o instanceof StructureSpawn && o.spawning);
-            if (spawns && VISUALS.SPAWN) {
+            const towers = _.find(look, o => o instanceof StructureTower);
+ +			if (towers) {
+ +				y += 0.4;
+ +			} else {
```
Comment (Spedwards) [inline visuals.js]: More pluses and tabs.

File: visuals.js [hunk 3]
```diff
@@ -359,6 +368,13 @@ module.exports = class Visuals {
         }
     }
     
+    static drawTowerInfo(tower) {
+ +		const vis = new RoomVisual();
+ +		const x = tower.pos.x + 1;
+ +		const y = tower.pos.y - 0.5;
+ +		vis.text(`Energy: ${tower.energy}/${tower.energyCapacity}`, x, y, {font: 0.4, align: 'left'});
+ +	}
```
Comment (Spedwards) [inline visuals.js]: Again, pluses and tabs.

File: visuals.js [hunk 4]
```diff
@@ -359,6 +368,13 @@ module.exports = class Visuals {
         }
     }
     
+    static drawTowerInfo(tower) {
+ +		const vis = new RoomVisual();
```
Comment (Spedwards) [inline visuals.js]: `new RoomVisual(tower.room.name);`

You don't want to be creating global visuals.

**Comment by karlthepagan** [comment]
Will try to add this to canary.

**Comment by Perrytheplatypus43** [comment]
@Spedwards I was just wondering why you shortened 'Energy' to 'E', is it because the visual will be to big and it might be over lapping with structure so it will be hard to read?

**Comment by Spedwards** [comment]
@Perrytheplatypus43 

1. It's consistent with Weakest Structure, and Controller texts.
2. It's shorter so lower chance of overlap with other visuals.

**Comment by SEATURTLEKING** [comment]
maybe this visual can be one of the top visuals, not next to the tower in tiny print

**Comment by Spedwards** [comment]
@SEATURTLEKING That's not only inconsistent, but will clog your screen up once you have all 6 towers.

**Comment by Spedwards** [comment]
@karlthepagan @Perrytheplatypus43 This is good to merge now I believe.

**Pull request merged**